src/main/java/ui/UI.java

```diff
@@ -134,13 +135,18 @@ public void printAllocatingMessage() {
      * The table includes student IDs and respective allocation outcomes.
      *
      * @param studentList ArrayList of Student objects to be printed
+     * @throws SEPEmptyException If the student list is null or empty,
+     *     indicating that there are no students to generate.
      */
-    public void generateReport(ArrayList<Student> studentList) {
+    public void generateReport(ArrayList<Student> studentList) throws SEPEmptyException {
         AsciiTable at = new AsciiTable();
         at.addRule();
         at.addRow("Student", "University Granted");
         at.addRule();
 
+        if (studentList == null || studentList.isEmpty()) {
```

Comment:
Edge cases should be handled at the start of the function.

Reply:
oops, not sure why the order changed, fixed.